### ISSUE TITLE
pdm run does not activate the project's environment when running the command

### ISSUE THREAD

**Issue by pfmoore** [pdm-project/pdm#1650]
- [X] I have searched the issue tracker and believe that this is not a duplicate.

## Steps to reproduce

```
❯ pdm run -v python -c "import sys; print(sys.executable)"
C:\Work\Scratch\Python\repro\.venv\Scripts\python.exe
❯ pdm run -v py -c "import sys; print(sys.executable)"
C:\Users\Gustav\AppData\Local\Programs\Python\Python311\python.exe
```

## Actual behavior

Note that running `python` shows the copy in the venv, but running `py` shows the system Python.

## Expected behavior

When the requested command is run, the `VIRTUAL_ENV` environment variable is set and the environment `bin`/`Scripts` directory is added to the start of `PATH`, in the same way as would happen with an activated virtual environment.

## Environment Information

```bash
❯ pdm info; pdm info --env
PDM version:
  2.4.1
Python Interpreter:
  C:\Work\Scratch\Python\repro\.venv\Scripts\python.exe (3.11)
Project Root:
  C:/Work/Scratch/Python/repro
Project Packages:
  None
{
  "implementation_name": "cpython",
  "implementation_version": "3.11.0",
  "os_name": "nt",
  "platform_machine": "AMD64",
  "platform_release": "10",
  "platform_system": "Windows",
  "platform_version": "10.0.22621",
  "python_full_version": "3.11.0",
  "platform_python_implementation": "CPython",
  "python_version": "3.11",
  "sys_platform": "win32"
}
```

**Comment by pawamoy** [comment]
IIUC, the `py` executable on Windows is a wrapper that has its own rules for finding and calling a Python interpreter installed on the system, so this is kinda expected :thinking:

**Comment by pfmoore** [comment]
It is. It checks for `VIRTUAL_ENV`, which is why that should be set to allow it to work as expected, IMO. (I added that functionality to it precisely so that situations like this would work...)

**Comment by pawamoy** [comment]
Oh, I see, thanks, didn't know that :+1:

**Comment by pfmoore** [comment]
I did a bit more digging. It looks like the virtual environment is added to `PATH`, so all that is really needed is to add the `VIRTUAL_ENV` environment variable when needed. I've had a look at the code, but I can't immediately see how to do this - I've found the place in the code, but I'm not very clear on how I can check if an `Environment` is a virtual environment (as opposed to something using the PEP 582-like `__pypackages__` mechanism). If anyone can give me a hint about that, I can maybe try to put together a PR myself.
